mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLAttributes.td

```diff
@@ -13,11 +13,18 @@ include "mlir/IR/EnumAttr.td"
 
 include "SYCLBase.td"
 
-class SYCL_I32EnumAttr<string name, string summary, list<I32EnumAttrCase> cases>
+class SYCL_I32Enum<string name, string summary, list<I32EnumAttrCase> cases>
     : I32EnumAttr<name, summary, cases> {
+  let genSpecializedAttr = 0;
   let cppNamespace = "::mlir::sycl";
 }
 
+class SYCL_I32EnumAttr<string name, string summary, string mnemonic,
+                       list<I32EnumAttrCase> cases>
+    : EnumAttr<SYCL_Dialect, SYCL_I32Enum<name, summary, cases>, mnemonic> {
+  let assemblyFormat = "`<` $value `>`";
+}
```

Comment:
We now generate both enums and attributes for each case. We will be using enums for now, as migration to attributes require many changes.